capstone/src/main/java/com/google/sps/data/Classroom.java

```diff
@@ -1,94 +1,100 @@
+        this.entity.setProperty(TEACHER, teacher.getUserKey());
+        this.entity.setProperty(SUBJECT, subject);
+        this.entity.setProperty(STUDENTS, new ArrayList<>());
+        this.entity.setProperty(FORMS, new ArrayList<>());
     }
 
-    public void setKey(Key key) {
-        this.key = key;
+    public void setTeacher(User teacher) { 
+        this.entity.setProperty(TEACHER, teacher.getUserKey());
     }
-
-    // Getters
-    public Teacher getTeacher() {
-        return this.teacher;
+    
+    public Key getTeacher() {
+        return (Key) this.entity.getProperty(TEACHER);
     }
 
     public String getSubject() {
-        return this.subject;
+        return (String) this.entity.getProperty(SUBJECT);
     }
 
-    public List<Student> getAllStudents() {
-        return this.students;
+    public Entity getClassroomEntity() {
+        return this.entity;
     }
 
-    public List<Form> getForms() {
-        return this.forms;
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllStudents() {
+        return (ArrayList<Key>) this.entity.getProperty(STUDENTS);
     }
 
-    public Key getKey() {
-        assert this.key.getId() != -1: "The key doesn't exist.";
-        return this.key;
+    @SuppressWarnings("unchecked")
+    public List<Key> getAllForms() { 
+        return (ArrayList<Key>) this.entity.getProperty(FORMS); 
     }
+    
+    @SuppressWarnings("unchecked")
+    public void addStudent(User student) {
+        List<Key> students = (ArrayList<Key>) this.entity.getProperty(STUDENTS);
+        students.add(student.getUserKey());
 
-    // Database
-    public void addStudent(Student student) { 
-        students.add(student);
+        this.entity.setProperty(STUDENTS, students);
     }
 
+    @SuppressWarnings("unchecked")
+    public void removeStudent(User student) {
+        List<Key> students = (ArrayList<Key>) this.entity.getProperty(STUDENTS);
+        students.remove(student.getUserKey());
+
+        this.entity.setProperty(STUDENTS, students);
+    }
+
+    @SuppressWarnings("unchecked")
     public void addForm(Form form) {
-        forms.add(form);
+        List<Key> forms = (ArrayList<Key>) this.entity.getProperty(FORMS);
+        forms.add(form.getFormKey());
+
+        this.entity.setProperty(FORMS, forms);
     }
 
-    public void removeStudent(Student student) {
-        this.students.remove(student);
+    @SuppressWarnings("unchecked")
+    public void removeForm(Form form) {
+        List<Key> forms = (ArrayList<Key>) this.entity.getProperty(FORMS);
+        forms.remove(form.getFormKey());
+
+        this.entity.setProperty(FORMS, forms);
     }
 
-    public boolean isStudentInClass(Student student) {
+    @SuppressWarnings("unchecked")
+    public boolean isStudentInClass(User student) {
+        List<Key> students = (ArrayList<Key>) this.entity.getProperty(STUDENTS);
+
         assert students.size() > 0: "There are no students in this class!";
```

Comment:
Asserts should be used when you think that the case only happens when a programmer has made an error. So if you have something like:
// Hey developers, you should only call this with positive numbers
public double sqrt(double input) {}

Then that would be an scenario where you would assert that the input is positive. In this case, classrooms legitimately can have no students and thats normal, it just means that you always return false. (But you dont actually need a separate check for that, since the rest of your logic applies equally well to an empty list